mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -784,10 +784,13 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
                 mPendingCall = null;
                 mInitState = InitState.DONE;
 
+                String defaultSitemapName =
+                        mSettings.getString(Constants.PREFERENCE_SITEMAP_NAME, "");
+
                 // OH1 returns XML, later versions return JSON
                 List<OpenHABSitemap> result = mOpenHABVersion == 1
-                        ? loadSitemapsFromXml(responseBody)
-                        : loadSitemapsFromJson(responseBody);
+                        ? loadSitemapsFromXml(responseBody, defaultSitemapName)
+                        : loadSitemapsFromJson(responseBody, defaultSitemapName);
```

Comment:
I don't like default sitemap name being passed to the load methods, as it's not required for loading. Suggestion: leave sorting out of the load methods completely and instead ...